lib/cloud_controller/dea/instances_reporter.rb

```diff
@@ -13,16 +13,23 @@ def all_instances_for_app(app)
 
       def number_of_starting_and_running_instances_for_process(app)
         return 0 unless app.started?
-        return 0 if app.staging_failed?
+        return 0 if app.current_droplet.nil?
         health_manager_client.healthy_instances(app)
       end
 
-      def number_of_starting_and_running_instances_for_processes(apps)
-        apps_without_running_instances = apps.select { |app| !app.started? || app.staging_failed? || app.pending? }
-        apps_without_running_instances.inject(
-          healthy_instances_bulk(apps - apps_without_running_instances)
-        ) do |result, app|
-          result.update(app.guid => 0)
+      def number_of_starting_and_running_instances_for_processes(processes)
```

Comment:
I would rather leave this all as one commit since wraps up all of the DEA query logic into one commit. If it was based off of endpoints, then there would be number of commit.